src/Raven.Server/Smuggler/Documents/SingleShardDatabaseSmuggler.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Documents.Smuggler;
+using Raven.Client.ServerWide;
+using Raven.Client.Util;
+using Raven.Server.Documents;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Commands;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Smuggler.Documents.Data;
+using Raven.Server.Utils;
+using Sparrow.Json;
+
+namespace Raven.Server.Smuggler.Documents
+{
+    internal class SingleShardDatabaseSmuggler : DatabaseSmuggler
+    {
+        private readonly TransactionContextPool _serverContextPool;
+        private readonly DatabaseRecord _shardedRecord;
+        private readonly int _index;
+        private readonly bool _lastShard;
+        public SingleShardDatabaseSmuggler(DocumentDatabase database, ISmugglerSource source, ISmugglerDestination destination, SystemTime time, 
+            JsonOperationContext context, DatabaseSmugglerOptionsServerSide options, SmugglerResult result = null, 
+            Action<IOperationProgress> onProgress = null, CancellationToken token = default) : 
+            base(database, source, destination, time, context, options, result, onProgress, token)
+        {
+            _serverContextPool = database.ServerStore.ContextPool;
+            _shardedRecord = _source.GetShardedDatabaseRecordAsync().Result;
```

Comment:
this about removing this or moving that to InitializeAsync

Reply:
going to remove this